action/protocol/poll/protocol.go

```diff
@@ -396,6 +421,22 @@ func (p *governanceChainCommitteeProtocol) getGravityHeight(height uint64) (uint
 	return p.electionCommittee.HeightByTime(blkTime)
 }
 
+func (p *governanceChainCommitteeProtocol) mergeDelegates(cand state.CandidateList, votes []*types.Vote) error {
+	// as of now, native staking does not have register contract, only voting/staking contract
+	// it is assumed that all votes done on native staking target for delegates registered on Ethereum
+	// votes cast to all outside address will not be counted and simply ignored
+	for i, c := range cand {
+		for _, v := range votes {
+			if bytes.Compare(v.Candidate(), c.CanName) == 0 {
+				c.Votes.Add(c.Votes, v.Amount())
```

Comment:
```suggestion
				vs = big.NewInt(0).add(c.Votes, v.Amount())
				cand[i] = &state.Candidate{c.Address, vs, c.RewardAddress, c.CanName}
```

Avoid mutating the existing var?

Reply:
Agree, let's return a new list

Reply:
@zjshen14   in the first version review, you suggested c.Votes.add(v.Amount())? that is why change to c.Votes.Add(c.Votes, v.Amount()) 

c and c.Votes are both pointer, and i added test to verify this is correct, so should be fine i think